bot/data/class_repository.py

```diff
@@ -0,0 +1,151 @@
+        for semester in await self.get_all_semesters():
+            if semester.start_date() <= current_datetime <= semester.end_date():
+                return semester
+        return None
+
+    async def get_next_semester(self) -> ClassSemester | None:
+        """
+        Gets the next semester, if one exists.
+        """
+        current_datetime = datetime.datetime.utcnow()
+        for semester in await self.get_all_semesters():
+            if current_datetime < semester.start_date():
+                return semester
+        return None
+
+    async def get_unarchived_classes(self) -> list[ClassChannel]:
+        """
+        Gets a list of class channels that are currently marked as unarchived.
+        """
+        async with aiosqlite.connect(self.resolved_db_path) as db:
+            cursor = await db.execute("SELECT * FROM ClassChannel WHERE class_archived IS FALSE")
+            return [ClassChannel(**d) for d in await self.fetcthall_as_dict(cursor)]
+
+    async def get_semester_classes(self, semester: ClassSemester, unarchived_only: bool = True) -> list[ClassChannel]:
+        """
+        Gets a list of class channels for a specific semester, with the ability to
+        only fetch the ones that are currently marked as unarchived.
+        """
+        statement = 'SELECT * FROM ClassChannel WHERE semester_id = ?' \
+                    + ' AND class_archived = FALSE' if unarchived_only else ''
+        async with aiosqlite.connect(self.resolved_db_path) as db:
+            cursor = await db.execute(statement, (semester.semester_id,))
+            return [ClassChannel(**d) for d in await self.fetcthall_as_dict(cursor)]
+
+    async def search_semester(self, semester_id: str) -> ClassSemester | None:
+        """
+        Searches for a semester with the given semester ID.
+        Note that this does not search for the semester name, which is different.
+        ex: Semester Name: Spring 2023, Semester ID: sp2023
+        """
+        async with aiosqlite.connect(self.resolved_db_path) as db:
+            cursor = await db.execute("SELECT * FROM ClassSemester WHERE semester_id = ?", (semester_id,))
+            dictionary = await self.fetcthone_as_dict(cursor)
+            if not len(dictionary):
+                return None
+            return ClassSemester(**dictionary)
+
+    async def search_class(self, prefix: str, num: int, prof: str) -> ClassChannel | None:
+        """
+        Searches for a registered class given the class abbreviation, class number, and class instructor.
+        """
+        async with aiosqlite.connect(self.resolved_db_path) as db:
+            cursor = await db.execute(
+                """ SELECT * FROM ClassChannel 
+                    WHERE class_prefix = ? 
+                    AND class_number = ? 
+                    AND class_professor = ?""", (prefix, num, prof)
+            )
+            dictionary = await self.fetcthone_as_dict(cursor)
```

Comment:
fetcthone

Reply:
When I was first starting this PR I was really confused as to why it was named this as well. I'll probably go back and make a better name for them...

Reply:
fetchone ?

Reply:
more like fetch_first 😈